### PR TITLE
Fix unbounded write

### PR DESCRIPTION
**Pull request by asvrada**
https://github.com/deislabs/mystikos/security/code-scanning?query=%22unbounded+write%22+is%3Aopen+branch%3Amain+autofilter%3Atrue+

### CODE REVIEW

File: host/maps.c [hunk 1]
```diff
@@ -103,7 +103,7 @@ int myst_maps_load(myst_maps_t** maps_out)
 
         n = sscanf(
             line,
-            "%lx-%lx %c%c%c%c %lu %u:%u %lu %s",
+            "%lx-%lx %c%c%c%c %lu %u:%u %lu %4096s",
```
Comment (radhikaj) [inline host/maps.c:106]: where does this 4096 come from

Comment (asvrada) [reply host/maps.c:106]: The max size for path is 4096 bytes

Comment (radhikaj) [reply host/maps.c:106]: > The path variable is of 4096 bytes in size

Is there a #define we can use instead?

Comment (mikbras) [reply host/maps.c:106]: Hmm? I wish there were a way without hardcoding the path length.

Comment (mikbras) [reply host/maps.c:106]: Okay, I verified that the following works:

```
        int path_offset;

        n = sscanf(
            line,
            "%lx-%lx %c%c%c%c %lu %u:%u %lu %n",
            &start,
            &end,
            &r,
            &w,
            &x,
            &f,
            &o,
            &minor,
            &major,
            &inode,
            &path_offset);

        if (n < 9)
        {
            ret = -ENOSYS;
            goto done;
        }

        *path = '\0';
        strncat(path, &line[path_offset], PATH_MAX);
```

Comment (mikbras) [reply host/maps.c:106]: Also, you probably want to check whether ``streln(*line[path_offset]) is bigger than ``PATH_MAX``.

**Review by mikbras — APPROVED** [review]
Thank you for doing this!

**Pull request merged**